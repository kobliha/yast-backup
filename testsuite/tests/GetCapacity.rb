# encoding: utf-8

# Backup module - testsuite
#
# GetCapacity function tests
#
# testedfiles: Backup.ycp
#
# $Id$
#
module Yast
  class GetCapacityClient < Client
    def main
      Yast.import "Testsuite"

      @readmap = {
        "target" => { "string" => "", "size" => -1, "tmpdir" => "/tmp" },
        "proc"   => { "cpuinfo" => { "value" => { "0" => { "flags" => "" } } } },
        "probe"  => {
          "has_smp" => false,
          "is_uml"  => false,
          "cdrom"   => { "manual" => [] }
        }
      }

      @execmap = {
        "target" => {
          "bash_output" => {
            "exit"   => 0,
            "stderr" => "",
            "stdout" => "charmap=\"ISO-8859-1\"\n"
          }
        }
      }

      Testsuite.Init([@readmap, {}, @execmap], nil)

      Yast.import "Pkg" # override packamanager
      Yast.import "Backup"

      Testsuite.Dump(Backup.GetCapacity(nil, nil))
      Testsuite.Dump(Backup.GetCapacity([], :s))
      Testsuite.Dump(Backup.GetCapacity([{ "capacity" => 1440000 }], :s))

      Testsuite.Dump(
        Backup.GetCapacity(
          [
            {
              "label"    => "Floppy 1.44 MB",
              "symbol"   => :fd144,
              "capacity" => 1440000
            },
            {
              "label"    => "Floppy 1.2 MB",
              "symbol"   => :fd12,
              "capacity" => 1200000
            }
          ],
          :fd12
        )
      )

      nil
    end
  end
end

Yast::GetCapacityClient.new.main
