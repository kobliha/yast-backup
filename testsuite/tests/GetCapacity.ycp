// Backup module - testsuite
//
// GetCapacity function tests
//
// $Id$
//

{

include "backup/functions.ycp";
include "testsuite.ycp";

// proc.cpuinfo.value.0.flags, target.string - used in Kernel::
// probe.has_smp - Arch::
// probe.cdrom.manual - StorageDevices::
// target.size - AutoInstall::

TESTSUITE_INIT([ $[ "target" : $["string" : "", "size" : -1, "tmpdir" : "/tmp" ],
    "proc" : $["cpuinfo" : $[ "value" : $["0" : $[ "flags" : ""]]]],
    "probe" : $["has_smp" : false, "cdrom" : $["manual" : []]]  ], [], [] ] ,nil);

import "Pkg";	// override packamanager
import "Backup";

DUMP( Backup::GetCapacity(nil, nil) );
DUMP( Backup::GetCapacity([ ], `s) );
DUMP( Backup::GetCapacity(["no map"], nil) );
DUMP( Backup::GetCapacity([
    $[
	"capacity" : 1440000
    ]
], `s) );

DUMP( Backup::GetCapacity([
    $[
	"label" : _("Floppy 1.44 MB"),
	"symbol" : `fd144,
	"capacity" : 1440000
    ],
    $[
	"label" : _("Floppy 1.2 MB"),
	"symbol" : `fd12,
	"capacity" : 1200000
    ]
], `fd12) );

}

