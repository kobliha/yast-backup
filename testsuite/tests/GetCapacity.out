Read	.target.tmpdir "/tmp"
Dump	nil
Dump	nil
Dump	nil
Dump	nil
Dump	1200000
